### PR TITLE
Add knowledge graph components

### PR DESCRIPTION
**Pull request by XinyaoWa**
## Description

Add Knowledge graph components, which is based on Neo4j and can work as an enhanced retriever

## Issues

`n/a`.

## Type of change
- [ ] New feature (non-breaking change which adds new functionality)

## Dependencies
neo4j
langchain
tiktoken
langchain_openai
langchainhub
langchain_community
sentence-transformers

## Tests

`n/a`.

### CODE REVIEW

File: comps/knowledgegraphs/langchain/knowledge_graph.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+
+    ## Connect to Neo4j
+    neo4j_endpoint = os.getenv("NEO4J_ENDPOINT", "neo4j://localhost:7687")
+    neo4j_username = os.getenv("NEO4J_USERNAME", "neo4j")
+    neo4j_password = os.getenv("NEO4J_PASSWORD", "neo4j")
+    graph = Neo4jGraph(url=neo4j_endpoint, username=neo4j_username, password=neo4j_password)
+
+    ## keep for multiple tests, will remove later
+    graph.query("MATCH (n) DETACH DELETE n")
+    import_url = "https://gist.githubusercontent.com/tomasonjo/08dc8ba0e19d592c4c3cde40dd6abcc3/raw/e90b0c9386bf8be15b199e8ac8f83fc265a2ac57/microservices.json"
```
Comment (xuechendi) [inline comps/knowledgegraphs/langchain/knowledge_graph.py:117]: What is this url used for?

**Comment by xuechendi** [comment]
@XinyaoWa , The PR looks good, to make it clearer for other reviewer, please also provide info related to 
1. a workflow diagram
2. screenshot for both rag query and cypher query

File: comps/knowledgegraphs/langchain/knowledge_graph.py (hunk 1 above)
Comment (XinyaoWa) [reply comps/knowledgegraphs/langchain/knowledge_graph.py]: It's original for downloading example json data. Have refined this, remove the url and add the example data to comps/knowledgegraphslangchain/data/microservice.json

**Comment by XinyaoWa** [comment]
@lvliang-intel  Hi Lvliang, please have a review for this PR, thanks a lot!

**Comment by XinyaoWa** [comment]
> @XinyaoWa, Please resolve the conflict.

Resolved.

**Review by lvliang-intel — APPROVED** [review]

**Pull request merged**